### PR TITLE
Fix Certificate Thumbprints

### LINKED ISSUE #94: `calcuateHash` strips two leading zeros resulting in incorrect hash value
Please, see the [#53 in `dgc-lib`](https://github.com/eu-digital-green-certificates/dgc-lib/issues/53) as the exact same bug is present in [here](https://github.com/eu-digital-green-certificates/dgc-gateway/blob/1ebe5ad80506da34abf96f4cf763c356384e3c71/src/main/java/eu/europa/ec/dgc/gateway/restapi/filter/CertificateAuthenticationFilter.java#L84) on the line [110](https://github.com/eu-digital-green-certificates/dgc-gateway/blob/1ebe5ad80506da34abf96f4cf763c356384e3c71/src/main/java/eu/europa/ec/dgc/gateway/restapi/filter/CertificateAuthenticationFilter.java#L110).

### PR DESCRIPTION
**Pull request by f11h**
This PR fixes the Certificate Thumbprint calculation. (Missing leading "00")

This PR also includes a Liquibase Changeset to repair invalid entities in the database.

closes #94

### CODE REVIEW

**Review by mschulte-tsi — APPROVED** [review]

**Review by ascheibal — APPROVED** [review]

**Comment by quapka** [comment]
@mschulte-tsi @ascheibal are my comments on the SQL "fix", not a concern?

**Comment by f11h** [comment]
which comments? I cannot see anything :(

**Comment by quapka** [comment]
@f11h oh no, sorry! I haven't submitted the comments as a review, you should see them now. Thought others _can_ see them. :frowning_face:

File: src/main/resources/db/changelog/fix-certificate-thumbprints.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.3.xsd"
+        objectQuotingStrategy="QUOTE_ONLY_RESERVED_WORDS">
+    <changeSet id="fix-thumbprints" author="f11h">
+
+        <sql dbms="h2, mysql">
+            UPDATE trusted_party SET thumbprint = concat('00', thumbprint) WHERE length(thumbprint) = 62;
```
[the remark this answers is not in the document]
Comment (f11h) [reply src/main/resources/db/changelog/fix-certificate-thumbprints.xml:11]: In general you are right - Your solution is smart, I like it.
But we do not have any certificate with a bad hash on PRD. Only on ACC we have one invalid cert with a leading zero-byte. So my "Not-Broadband-Fix" is enough for now :)

**Pull request merged**